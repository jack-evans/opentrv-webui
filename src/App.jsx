import React, { Component } from 'react'
import Header from './components/Header/Header'
import './App.scss'
import SideBarMenu from './components/SideBarMenu/SideBarMenu'
import MainContent from './components/MainContent/MainContent'

class App extends Component {
  constructor (props) {
    super(props)
    this.state = {
      sideBarMenuOpen: false
    }

    this.sideBarMenuEvent = this.handleViewSideBarMenu.bind(this)
  }

  handleViewSideBarMenu () {
    this.setState({
      sideBarMenuOpen: !this.state.sideBarMenuOpen
    })
  }

  /**
   * render method
   *
   * Creates the html for the App Component Class
   * @returns {HTML} - App Component
   */
  render () {
    return (
      <div className='App'>
        <Header onClick={this.sideBarMenuEvent} isOpen={this.state.sideBarMenuOpen} />
        <SideBarMenu isOpen={this.state.sideBarMenuOpen} />
        <MainContent />
      </div>
    )
  }
}

export default App
