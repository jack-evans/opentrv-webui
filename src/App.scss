.App {
  text-align: center;

  .App__intro {
    font-size: large;
  }

  .App__menu-icon {
    fill: #fff;
  }

  .App__overflow-menu {
    display: flex;
    justify-content: center;
    padding-top: 50px;
  }
}
