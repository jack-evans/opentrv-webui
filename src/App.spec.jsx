import React from 'react'
import App from './App'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('App.jsx', () => {
  'use strict'

  it('renders without crashing', () => {
    expect(JSON.stringify(
      Object.assign({}, App, { _reactInternalInstance: 'censored' })
    )).toMatchSnapshot()
  })
})
