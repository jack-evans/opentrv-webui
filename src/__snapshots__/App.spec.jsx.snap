// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.jsx renders without crashing 1`] = `"{\\"_reactInternalInstance\\":\\"censored\\"}"`;
